core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -65,24 +65,18 @@ class LogSegment private[log] (val log: FileRecords,
     val reachedRollMs = timeWaitedForRoll(rollParams.now, rollParams.maxTimestampInMessages) > rollParams.maxSegmentMs - rollJitterMs
     size > rollParams.maxSegmentBytes - rollParams.messagesSize ||
       (size > 0 && reachedRollMs) ||
-      offsetIndex.isFull || timeIndex.isFull || !canConvertToRelativeOffset(rollParams.maxOffsetInMessages)
+      offsetIndex.get.isFull || timeIndex.get.isFull || !canConvertToRelativeOffset(rollParams.maxOffsetInMessages)
```

Comment:
You can introduce a `offsetIndex` method that calls `get` on the field so that most code stays the same.

Reply:
Good point. Thanks. Done.